packages/astro/src/vite-plugin-astro-server/index.ts

```diff
@@ -104,31 +104,13 @@ async function handle404Response(
 	req: http.IncomingMessage,
 	res: http.ServerResponse
 ) {
-	const site = config.site ? new URL(config.base, config.site) : undefined;
+	const site = config.site ? new URL(config.site) : undefined;
 	const devRoot = site ? site.pathname : '/';
 	const pathname = decodeURI(new URL(origin + req.url).pathname);
 	let html = '';
 	if (pathname === '/' && !pathname.startsWith(devRoot)) {
 		html = subpathNotUsedTemplate(devRoot, pathname);
 	} else {
-		// HACK: redirect without the base path for assets in publicDir
```

Comment:
No longer needed since Vite's dev server is handling `public` assets internally in now